cmd/flow-db.go

```diff
@@ -0,0 +1,223 @@
+		log.Fatal(err)
+	}
+	file.Close()
+	log.Println("flows.db created")
+	// Open SQLite database
+	db, err := sql.Open("sqlite3", flowsDB)
+	if err != nil {
+		log.Errorf("Error opening database: %v", err.Error())
+		return
+	}
+	defer db.Close()
+
+	// Create messages table if not exists
+	err = createFlowsDBTable(db)
+	if err != nil {
+		log.Errorf("Error creating table: %v", err.Error())
+		log.Fatal(err)
+	}
+	for fp := range flowPackets {
+		handleFlowsDBConnection(db, fp.GenericMap.Value)
+	}
+}
+
+func createFlowsDBTable(db *sql.DB) error {
+	createFlowsTableSQL := `CREATE TABLE flow (
+		"DnsErrno" INTEGER,
+		"Dscp" INTEGER,
+		"DstAddr" TEXT ,
+		"DstPort" INTEGER,
+		"Interface" TEXT,
+		"Proto" INTEGER,
+		"SrcAddr" TEXT,
+		"SrcPort" INTEGER,
+		"Bytes" INTEGER,
+		"Packets" INTEGER,
+		"PktDropLatestDropCause" TEXT,
+		"PktDropBytes" INTEGER,
+		"PktDropPackets" INTEGER,
+		"DnsId" INTEGER,
+		"DnsFlagsResponseCode" TEXT,
+		"DnsLatencyMs" TIMESTAMP,
+		"TimeFlowRTTNs" TIMESTAMP
+	  );` // SQL Statement for Create Table
+
+	log.Println("Create flows table...")
+	statement, err := db.Prepare(createFlowsTableSQL) // Prepare SQL Statement
+	if err != nil {
+		if err.Error() == "table flow already exists" {
+			return nil
+		}
+		log.Errorf("Error prepare table: %v", err.Error())
+		return err
+	}
+	_, err = statement.Exec() // Execute SQL Statements
+	if err != nil {
+		if err.Error() == "table flow already exists" {
+			return nil
+		}
+		log.Errorf("Error creating table: %v", err.Error())
+		return err
+	}
+
+	log.Println("flows table created")
+	return nil
+}
+
+func handleFlowsDBConnection(db *sql.DB, buf []byte) {
+	flow := config.GenericMap{}
+
+	// Unmarshal the JSON string into the flow object
+	err := json.Unmarshal(buf, &flow)
+	if err != nil {
+		log.Errorf("Error: %s", err)
+		return
+	}
+	// Insert message into database
+	var flowSQL string
+	if flow["PktDropPackets"] != 0 && flow["DnsId"] != 0 {
+		flowSQL =
+			`INSERT INTO flow(DnsErrno, Dscp, DstAddr, DstPort, Interface, Proto, SrcAddr, SrcPort, Bytes, Packets, PktDropLatestDropCause, PktDropBytes, PktDropPackets, DnsId, DnsFlagsResponseCode, DnsLatencyMs, TimeFlowRttNs) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`
+	} else if flow["PktDropPackets"] != 0 {
+		flowSQL =
+			`INSERT INTO flow(DnsErrno, Dscp, DstAddr, DstPort, Interface, Proto, SrcAddr, SrcPort, Bytes, Packets, PktDropLatestDropCause, PktDropBytes, PktDropPackets, TimeFlowRttNs) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`
+	} else if flow["DnsId"] != 0 {
+		flowSQL =
+			`INSERT INTO flow(DnsErrno, Dscp, DstAddr, DstPort, Interface, Proto, SrcAddr, SrcPort, Bytes, Packets, DnsId, DnsFlagsResponseCode, DnsLatencyMs, TimeFlowRttNs) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`
+	} else {
+		flowSQL =
+			`INSERT INTO flow(DnsErrno, Dscp, DstAddr, DstPort, Interface, Proto, SrcAddr, SrcPort, Bytes, Packets, TimeFlowRttNs) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`
+	}
```

Comment:
Would it be easier to inject missing values and run the same insert query all the time ?

Else an alternative could be to use a string builder and append fields conditionnally